queries/tact/textobjects.scm

```diff
@@ -0,0 +1,278 @@
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "function.inner" @_start @_end))
+
+; class.inner & outer
+; -------------------
+(struct
+  body: (struct_body
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "class.inner" @_start @_end)) @class.outer
+
+(message
+  body: (message_body
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "class.inner" @_start @_end)) @class.outer
+
+(contract
+  body: (contract_body
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "class.inner" @_start @_end)) @class.outer
+
+(trait
+  body: (trait_body
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "class.inner" @_start @_end)) @class.outer
+
+; attribute.inner & outer
+; -----------------------
+("@name"
+  "("
+  func_name: (func_identifier) @attribute.inner")") @attribute.outer
+
+(contract_attributes
+  ("@interface"
+    "("
+    (string) @attribute.inner
+    ")") @attribute.outer)
+
+(trait_attributes
+  ("@interface"
+    "("
+    (string) @attribute.inner
+    ")") @attribute.outer)
+
+(trait_attributes
+  ("@interface"
+    "("
+    (string) @attribute.inner
+    ")") @attribute.outer)
+
+; loop.inner & outer
+; ------------------
+(while_statement
+  body: (block_statement
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "loop.inner" @_start @_end)) @loop.outer
+
+(repeat_statement
+  body: (block_statement
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "loop.inner" @_start @_end)) @loop.outer
+
+(do_until_statement
+  body: (block_statement
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "loop.inner" @_start @_end)) @loop.outer
+
+(foreach_statement
+  body: (block_statement
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "loop.inner" @_start @_end)) @loop.outer
+
+; conditional.inner & outer
+; -------------------------
+(if_statement
+  consequence: (block_statement
+    "{"
+    (_) @_start @_end
+    (_)? @_end
+    "}")
+  (#make-range! "conditional.inner" @_start @_end)) @conditional.outer
+
+(if_statement
+  alternative: (else_clause
+    (block_statement
+      "{"
+      (_) @_start @_end
+      (_)? @_end
+      "}"))
+  (#make-range! "conditional.inner" @_start @_end)) @conditional.outer
+
+(if_statement) @conditional.outer
+
+; block.inner & outer
+; -------------------
+(_
+  (block_statement) @block.inner) @block.outer
+
+; call.inner & outer
+; ------------------
+(method_call_expression
+  arguments: (argument_list
+    "("
+    (_) @_start
+    (_)? @_end
+    ")")
+  (#make-range! "call.inner" @_start @_end)) @call.outer
+
+(static_call_expression
+  arguments: (argument_list
+    "("
+    (_) @_start
+    (_)? @_end
+    ")")
+  (#make-range! "call.inner" @_start @_end)) @call.outer
+
+(instance_expression
+  arguments: (instance_argument_list
+    "{"
+    (_) @_start
+    (_)? @_end
```

Comment:
Right now this query will match at most two inner argument nodes (same with the other similar ones): usually some sort of `(_)*` is needed; but if the argument list is always surrounded by brackets would it be possible to just use `#offset! 0 1 0 -1` here?